src/lxc/confile.c

```diff
@@ -403,6 +405,33 @@ static int set_config_net_link(const char *key, const char *value,
 	return ret;
 }
 
+static int set_config_net_l2proxy(const char *key, const char *value,
+				     struct lxc_conf *lxc_conf, void *data)
+{
+	struct lxc_netdev *netdev = data;
+	unsigned int val = 0;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_l2proxy(key, lxc_conf, data);
+
+	if (!netdev)
+		return -1;
+
+	if (lxc_safe_uint(value, &val) < 0)
+		return minus_one_set_errno(EINVAL);
```

Comment:
I think the `lxc_safe_uint()` functions will set errno correctly so better would be:
```c
ret = lxc_safe_uint();
if (ret < 0)
        return minus_...
```